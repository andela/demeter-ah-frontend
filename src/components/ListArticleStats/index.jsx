import React, { Fragment } from 'react';
import List from './list';

const ListArticleStats = ({ articles }) => (
  <Fragment>
    <div className="py-4 flex justify-center">
      <table className="w-full text-xs bg-white shadow-md rounded mb-4 overflow-scroll">
        <tbody>
          <tr className="border-b bg-gray-10">
            <th className="text-left p-3 px-5">Articles</th>
            <th className="text-center p-3 px-5">Likes</th>
            <th className="text-center p-3 px-5">Dislikes</th>
            <th className="text-center p-3 px-5">Stars</th>
            <th className="text-center p-3 px-5">Read</th>
            <th />
          </tr>
          <List articles={articles} />
        </tbody>
      </table>
    </div>
  </Fragment>
);

export default ListArticleStats;
