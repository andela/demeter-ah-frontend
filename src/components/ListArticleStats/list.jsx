import React, { Fragment } from 'react';
import Stars from '../Rating';

const List = ({ articles }) => {
  const showArticles = articles.length ? articles.map(article => (
    <Fragment key={article.id}>
      <tr className="border-b">
        <td className="p-3 px-5">
          {article.title}
        </td>
        <td className="p-3 px-5 text-center">{article.upVote.length}</td>
        <td className="p-3 px-5 text-center">{article.downVote.length}</td>
        <td className="p-3 px-5 text-center flex items-center justify-center">
          <Stars rating={article.rating} />
        </td>
        <td className="p-3 px-5 text-center">{article.reads}</td>
      </tr>
    </Fragment>
  ))
    : (
      <tr className="text-center">
        <td className="p-3 px-5 text-center" />
        <td className="p-3 px-5 text-center" />
        <td className="p-3 px-5 text-center">NO Available stats</td>
        <td className="p-3 px-5 text-center" />
        <td className="p-3 px-5 text-center" />
      </tr>
    );
  return (
    showArticles
  );
};

export default List;
