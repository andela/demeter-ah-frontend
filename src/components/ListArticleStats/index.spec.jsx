import React from 'react';
import { shallow } from 'enzyme';
import ListArticleStats from './index';

describe('ListArticleStats Component', () => {
  it('Should render without errors', () => {
    const component = shallow(<ListArticleStats />);
    expect(component).toMatchSnapshot();
  });
});
