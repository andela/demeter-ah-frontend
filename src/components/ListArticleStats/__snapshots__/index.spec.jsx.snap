// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListArticleStats Component Should render without errors 1`] = `
<Fragment>
  <div
    className="py-4 flex justify-center"
  >
    <table
      className="w-full text-xs bg-white shadow-md rounded mb-4 overflow-scroll"
    >
      <tbody>
        <tr
          className="border-b bg-gray-10"
        >
          <th
            className="text-left p-3 px-5"
          >
            Articles
          </th>
          <th
            className="text-center p-3 px-5"
          >
            Likes
          </th>
          <th
            className="text-center p-3 px-5"
          >
            Dislikes
          </th>
          <th
            className="text-center p-3 px-5"
          >
            Stars
          </th>
          <th
            className="text-center p-3 px-5"
          >
            Read
          </th>
          <th />
        </tr>
        <List />
      </tbody>
    </table>
  </div>
</Fragment>
`;
